pkg/clusteragent/clusteragent.go

```diff
@@ -0,0 +1,127 @@
+package clusteragent
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	defaultClusterAgentPort = "25000"
```

Comment:
Here too, would it be possible to get the port from the configuration instead of hardcoding it?